commit 87eb3ec818fa57bc49ce419785c56d72d9da1faa
Author: Ramalingam C <ramalingam.c@intel.com>
Date:   Thu Jan 18 11:18:06 2018 +0530

    drm/i915: II stage HDCP auth for repeater only
    
    Second stage of HDCP authentication required only if the HDCP
    sink is a repeater.
    
    This patch imposes above condition on second stage HDCP authentication.
    
    Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
    Signed-off-by: Sean Paul <seanpaul@chromium.org>
    Link: https://patchwork.freedesktop.org/patch/msgid/1516254488-4971-3-git-send-email-ramalingam.c@intel.com

diff --git a/drivers/gpu/drm/i915/intel_hdcp.c b/drivers/gpu/drm/i915/intel_hdcp.c
index e49ca7fe893f..c8e611af408e 100644
--- a/drivers/gpu/drm/i915/intel_hdcp.c
+++ b/drivers/gpu/drm/i915/intel_hdcp.c
@@ -493,7 +493,10 @@ static int intel_hdcp_auth(struct intel_digital_port *intel_dig_port,
 	 * on those as well.
 	 */
 
-	return intel_hdcp_auth_downstream(intel_dig_port, shim);
+	if (repeater_present)
+		return intel_hdcp_auth_downstream(intel_dig_port, shim);
+
+	return 0;
 }
 
 static